src/acceptor.rs

```diff
@@ -0,0 +1,712 @@
+                Err(_) => Default::default(),
+            }
+        }
+    }
+
+    /// Get the i'th in the list of signature schemes offered in the ClientHello.
+    /// This is useful in selecting a server certificate when there are multiple
+    /// available for the same server name, for instance when selecting
+    /// between an RSA and an ECDSA certificate.
+    ///
+    /// Parameters:
+    ///
+    /// accepted: The rustls_accepted to access.
+    /// i: Fetch the signature scheme at this offset.
+    ///
+    /// Returns:
+    ///
+    /// A TLS Signature Scheme from <https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-signaturescheme>
+    ///
+    /// This will be 0 in these cases:
+    ///   - i is greater than the number of available cipher suites.
+    ///   - accepted is NULL.
+    ///   - rustls_accepted_into_connection has already been called with `accepted`.
+    #[no_mangle]
+    pub extern "C" fn rustls_accepted_signature_scheme(
+        accepted: *const rustls_accepted,
+        i: usize,
+    ) -> u16 {
+        ffi_panic_boundary! {
+            let accepted: &Option<Accepted> = try_ref_from_ptr!(accepted);
+            let accepted = match accepted {
+                Some(a) => a,
+                None => return 0,
+            };
+            let hello = accepted.client_hello();
+            let signature_schemes = hello.signature_schemes();
+            if i < signature_schemes.len() {
+                signature_schemes[i].get_u16()
+            } else {
+                0
+            }
+        }
+    }
+
+    /// Get the i'th in the list of cipher suites offered in the ClientHello.
+    ///
+    /// Parameters:
+    ///
+    /// accepted: The rustls_accepted to access.
+    /// i: Fetch the cipher suite at this offset.
+    ///
+    /// Returns:
+    ///
+    /// A cipher suite value from <https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4.>
+    ///
+    /// This will be 0 in these cases:
+    ///   - i is greater than the number of available cipher suites.
+    ///   - accepted is NULL.
+    ///   - rustls_accepted_into_connection has already been called with `accepted`.
+    ///
+    /// Note that 0 is technically a valid cipher suite "TLS_NULL_WITH_NULL_NULL",
+    /// but this library will never support null ciphers.
+    #[no_mangle]
+    pub extern "C" fn rustls_accepted_cipher_suite(
+        accepted: *const rustls_accepted,
+        i: usize,
+    ) -> u16 {
+        ffi_panic_boundary! {
+            let accepted: &Option<Accepted> = try_ref_from_ptr!(accepted);
+            let accepted = match accepted {
+                Some(a) => a,
+                None => return 0,
+            };
+            let hello = accepted.client_hello();
+            let cipher_suites = hello.cipher_suites();
+            if i < cipher_suites.len() {
+                cipher_suites[i].get_u16()
+            } else {
+                0
+            }
```

Comment:
I'd maybe write this as a `match`? For example:

```rust
match cipher_suites.get(i) {
    Some(cs) => cs.get_u16(),
    None => 0,
}
```

Reply:
Ah, good catch. Fixed.